chromadb/utils/data_loaders.py

```diff
@@ -22,3 +23,123 @@ def _load_image(self, uri: Optional[URI]) -> Optional[Image]:
+    with open(filepath, "rb") as f:
+        return f.read()
+    
+def url_to_contents(
+        url: str, content_extractor=attrgetter('text'), *, params=None, **kwargs
+    ) -> FileContents:
+    import requests
+
+    response = requests.get(url, params=params, **kwargs)
+    response.raise_for_status()
+    return content_extractor(response)
+
+def pdf_file_text(
+        filepath: str, *, page_break_delim='---------------------------'
+    ) -> str:
+    from pypdf import PdfReader  # pip install pypdf
+
+    return page_break_delim.join(
+        page.extract_text() for page in PdfReader(filepath).pages
+    )
+
+# --------------------------- FileLoader ---------------------------
+
+class FileLoader(DataLoader[List[Optional[FileContents]]]):
+    """A DataLoader that loads a list of text files from a list of URIs.
+
+    By default, it loads text files from local files, given URIs that are full file paths.
+    You can specify a prefix thought (usually used to specify a root directory),
+    or a suffix (usually used to specify a file extension).
+
+    Further, you can specify a different `loader`, e.g. to load files from a remote URL, 
+    or to load binary files, or to load text from pdf files, or from S3, or a database, etc.
+
+    Example:
+
+    >>> rootdir = chromadb.__path__[0] + '/'
+    >>> file_loader_1 = FileLoader(prefix=rootdir)
+    >>> file_contents_1 = file_loader_1(['__init__.py', 'types.py'])
+    >>> len(file_contents_1)
+    2
+    >>> 'Embeddings' in file_contents_1[0]  # i.e. __init__.py contains the word 'Embeddings'
+    >>> 'from typing import' in file_contents_1[1]  # i.e. types.py contains the phrase 'from typing import'
+    
+    """
+    
+    def __init__(
+        self,
+        loader=load_text,
+        *,
+        prefix: str = "",
+        suffix: str = "",
+        max_workers: int = multiprocessing.cpu_count(),
+    ) -> None:
+        """
+        Args:
+            loader: A function that takes a (single) URI and returns its contents
+            prefix: A string to prepend to the URI
+            suffix: A string to append to the URI
+            max_workers: The maximum number of threads to use when loading the URIs
+        """
+        self._loader = loader
+        self._prefix = prefix
+        self._suffix = suffix
+        self._max_workers = max_workers
+
+    def _load_file(self, uri: Optional[URI]) -> Optional[FileContents]:
+        if uri is None:
+            return None
+        return self._loader(f"{self._prefix}{uri}{self._suffix}")
+
+    def __call__(self, uris: Sequence[Optional[URI]]) -> List[Optional[FileContents]]:
+        if isinstance(uris, str):
+            # To avoid a common mistake, we cast a string to a list of containing it
+            uris = [uris]
+        return vectorize(self._load_file, uris, max_workers=self._max_workers)
+        
+# add a few loaders as attributes, for convenience
+FileLoader.load_text = load_text
```

Comment:
I'm going to go for 

```python
from types import SimpleNamespace

loaders = SimpleNamespace(
    load_text=load_text,
    load_bytes=load_bytes,
    url_to_contents=url_to_contents,
    pdf_file_text=pdf_file_text,
)
```

But again, I hesitate to even do this, since now that the collection is outside of the context it's supposed to be used, 
(a) it might not be clear to the user that this is meant to be used with `FileLoader`
(b) a user still needs to import `loaders`, so not that much more convenient than importing loader function directly

If you vote for not gathering these helper functions in a collection at all, I'll concede with no hesitation.